nixos/modules/services/networking/hylafaxp.nix

```diff
@@ -0,0 +1,262 @@
+    ${if (config.services.hylafaxp.devs!={ "dummy" = ""; }) then (concatStringsSep ";" (zipListsWith (a: b: "cp " + a + " /var/spool/hylafax/etc/config." + b) _configDevs _devs)) else ""}
+    chmod 0755 /var/spool/hylafax
+    ls -A /var/spool/hylafax/ | grep -v -E 'recvq|log' | ${pkgs.gawk}/bin/awk '{print "/var/spool/hylafax/"$1}' | xargs chmod -R 0755
+    chmod 0755 /var/spool/hylafax/{recvq,log} # avoid overriding RecvFileMode and LogFileMode in /var/spool/hylafax/etc/config. See: hylafax-config(4)
+    ##
+    ## faxsetup mkfifos but it seems the services do it fine on their own.
+    ## kept commented out for reference or any future complications.
+    #rm -f ${pkgs.hylafaxp}/var/spool/hylafax/FIFO* 2>/dev/null
+    #mkfifo -m 0777 ${pkgs.hylafaxp}/var/spool/hylafax/FIFO ${if config.services.hylafaxp.devs != { "" = ""; } then ("${pkgs.hylafaxp}/var/spool/hylafax/FIFO." + (concatStringsSep " ${pkgs.hylafaxp}/var/spool/hylafax/FIFO." _devs)) else ""} 2>/dev/null
+    ##
+    ${if (config.services.hylafaxp.hfaxd.hosts != "") then "chmod 0600 /var/spool/hylafax/etc/hosts.hfaxd" else ""} #runtime enforced sercurity. see: hfaxd/User.c++:160
+    chown -R "uucp:uucp" "/var/spool/hylafax"
+  '';
+  _config = pkgs.writeText "config" config.services.hylafaxp.config;
+  _hosts_hfaxd = pkgs.writeText "hosts.hfaxd" config.services.hylafaxp.hfaxd.hosts;
+  _FaxDispatch = pkgs.writeText "FaxDispatch" config.services.hylafaxp.faxDispatch;
+  _pagesizes = pkgs.writeText "pagesizes" config.services.hylafaxp.pagesizes;
+  ## list of devs
+  _devs = mapAttrsToList (n: v: n) config.services.hylafaxp.devs;
+  ## list of per-dev config files
+  _configDevs = mapAttrsToList (n: v: pkgs.writeText n v) config.services.hylafaxp.devs;
+
+in
+{
+
+  ###### interface
+
+  options = {
+
+    services.hylafaxp.enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable HylaFAX+, a Fax Management System.
+      '';
+    };
+
+    services.hylafaxp.devs = mkOption {
+      default = { "dummy" = ""; };
+      example = { ttyACM0 = "/var/spool/hylafax/etc/config.ttyACM0 content"; ttyS1 = "/var/spool/hylafax/etc/config.ttyS1 content"; };
+      type = types.attrsOf types.lines;
+      description = ''
+        The ttyFOOBAR devices and their /var/spool/hylafax/etc/config.ttyFOOBAR files.
+        Run faxsetup to generate the files interactively.
+        Be sure to copy the results to configuration.nix!
+      '';
+    };
+
+    services.hylafaxp.config = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        The content of the /var/spool/hylafax/etc/config file.
+        Run faxsetup to generate one interactively.
+        Be sure to copy the results to configuration.nix!
+      '';
+    };
+
+    services.hylafaxp.pagesizes = mkOption {
+      type = types.lines;
+      default = ''
+        #	$Id: pagesizes.in 2 2005-11-11 21:32:03Z faxguy $
```

Comment:
Yeah it's the paper size you send.
HylaFAX usually distributes with Letter as a default but I've taken the liberty to use ISO A4 instead since that's what I, and presumably, most of the NixOS user base, uses.

Anyhow, Americans\Japanese will want to change this.